src/Diagnostics.ModelsAndUtils/Models/GeoMaster/AzureSupportCenterInsight.cs

```diff
@@ -0,0 +1,164 @@
+
+        public Text Description { get; set; }
+
+        public RecommendedAction RecommendedAction { get; set; }
+
+        public CustomerReadyContent CustomerReadyContent { get; set; }
+
+        public IEnumerable<Link> Links { get; set; }
+
+        public string ConfidenceLevel { get; set; }
+
+        public string Scope { get; set; }
+    }
+
+    public enum ImportanceLevel
+    {
+        SomeLevel
+    }
+
+    public class CustomerReadyContent
+    {
+        public string ArticleId { get; set; }
+
+        public Text Text { get; set; }
+    }
+
+    public class Text
+    {
+        public bool IsMarkdown { get; set; }
+
+        public string Value { get; set; }
+    }
+
+    public class Link
+    {
+        public int Type { get; set; }
+
+        public string Text { get; set; }
+
+        public string Uri { get; set; }
+    }
+
+    public class RecommendedAction
+    {
+        public Guid Id { get; set; }
+
+        public Text Text { get; set; }
+    }
+
+    public static class AzureSupportCenterInsightUtilites
+    {
+        public static AzureSupportCenterInsight CreateInsight<TResource>(Insight insight, OperationContext<TResource> context, Definition detector)
+            where TResource : IResource
+        {
+
+            // Logic in line before would have to be updated if we added more resource types. 
+            // TODO: May make sense to have resource type enums contain the resource type string
+            var resourceTypeString = context.Resource.ResourceType == ResourceType.App ? "sites" : "hostingEnvironments";
+            var applensPath = $"subscriptions/{context.Resource.SubscriptionId}/resourceGroups/{context.Resource.ResourceGroup}/{resourceTypeString}/{context.Resource.Name}/detectors/{detector.Id}";
+
+            var recommendedAction = GetTextObjectFromData("recommended action", insight.Body);
+            var description = GetTextObjectFromData("description", insight.Body);
+            var customerReadyContent = GetTextObjectFromData("customer ready content", insight.Body);
+
+            var supportCenterInsight = new AzureSupportCenterInsight()
+            {
+                Id = GetDetectorGuid(detector.Id),
+                Title = insight.Message,
+                ImportanceLevel = ImportanceLevel.SomeLevel,
+                InsightFriendlyName = insight.Message,
+                IssueCategory = detector.Name.ToUpper(),
+                IssueSubCategory = detector.Name,
+                Description = description,
+                Links = new List<Link>()
+                    {
+                        new Link()
+                        {
+                            Type = 0,
+                            Text = "Applens Link",
+                            Uri = $"https://applens.azurewebsites.net/{applensPath}"
+                        }
+                    },
+                RecommendedAction = recommendedAction == null ? null : 
```

Comment:
nit : you can use ??

Reply:
too used to typescript where they don't have that!

Reply:
Actually for these recommendedAction is a Text object but RecommendedAction is a RecommendedAction object so I cannot use the ?? operator, as the two things being compared have to be of the same type.